addons/spectrum/functions/fnc_drawSpectrumLine.sqf

```diff
@@ -0,0 +1,56 @@
+#include "script_component.hpp"
+/*/////////////////////////////////////////////////
+Author: Landric
+			   
+File: fnc_drawSpectrumLine.sqf
+Parameters: colour - string corresponding to a colour in CfgMarkerColors
+Return: 
+
+Script to automatically draw a line on the map in the direction they are facing, labelled with the current freq. of the Spectrum Device
+Player must have the Spectrum Device equipped, be in "GUNNER" (i.e. zoomed-in) view, and have a GPS (or UAV terminal) assigned
+
+*///////////////////////////////////////////////
+
+params [["_colour", "ColorBlack", [""]]];
+
+if(!(
+		GVAR(spectrumAutoline) &&
+		(currentWeapon player) isKindOf ["hgun_esd_01_F", configFile >> "CfgWeapons"] &&
+		(
+			(player getSlotItemName 612) isKindOf ["ItemGPS", configFile >> "CfgWeapons"] ||
+			(player getSlotItemName 612) isKindOf ["UavTerminal_base", configFile >> "CfgWeapons"]
+		) &&
+		cameraOn == player && cameraView == "GUNNER"
+	)
```

Comment:
Change this to lazy evaluation, so it returns on the first mismatch, instead of evaluating all constraints if the first fails.

https://community.bistudio.com/wiki/Code_Optimisation#Lazy_Evaluation 

I would consider adding a new function called ``player_is_in_spectrum_view`` or something similar. Then only check for the gun and view there, and make another function that checks for ``player_has_gps_capabilities`` and call those two with lazy evaluation here. 

The benefit would be that they could be reused in other places. Especially the "in spectrum view" would probably be reusable in future code, as I could imagine I would want to check that for other things.

Reply:
Have spun those out and used lazy-eval